### PR TITLE
Major revamp of the docs.

### PR DESCRIPTION
**Pull request by jeromekelleher**
This is a major rejigger of the documentation:

- README is much smaller and points to docs for details.
- Existing docs have been moved around slightly.
- Added a demo for the API for new users.
- Client has been updated to have default args.

I would hope to merge this quickly, and then tag v0.1.0a3. We then make a PyPI package and the docs should automatically be built on read-the-docs. We can then review the docs today and tomorrow and fix the bugs that we find.

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                # More VCFs
+            variantSet2/
+                chr1.bcf
+                chr1.bcf.csi
+                chr2.bcf
+                chr2.bcf.csi
+                # More VCFs
+        /reads/
+            readGroupSet1
+                # TODO fill in details for read data.
```
Comment (dcolligan) [inline docs/source/configuration.rst]: put in example BAM/BAI file lines

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Configuration file
+------------------
+
+The GA4GH reference server is a `Flask application <http://flask.pocoo.org/>`_
+and uses the standard `Flask configuration file mechanisms
+<http://flask.pocoo.org/docs/0.10/config/>`_. An example configuration file
+might look like::
+
+    DATA_SOURCE = "/path/to/data/root"
+    # TODO other example config
```
Comment (dcolligan) [inline docs/source/configuration.rst]: either put in other lines or remove TODO

**Comment by dcolligan** [comment]
+1 modulo nits

**Comment by jeromekelleher** [comment]
Okey dokey, nits addressed. 

Merge me?

**Pull request merged**